uaclient/log.py

```diff
@@ -9,3 +12,53 @@ class RedactionFilter(logging.Filter):
     def filter(self, record: logging.LogRecord):
         record.msg = util.redact_sensitive_logs(str(record.msg))
         return True
+
+
+class JsonArrayFormatter(logging.Formatter):
+    """Json Array Formatter for our logging mechanism
+    Custom made for Pro logging needs
+    """
+
+    default_time_format = "%Y-%m-%dT%H:%M:%S%z"
+    required_fields = (
+        "asctime",
+        "name",
+        "funcName",
+        "lineno",
+        "levelname",
+        "message",
```

Comment:
I think levelname should go earlier to make ERROR and WARNING more visible
```suggestion
        "asctime",
        "levelname",
        "name",
        "funcName",
        "lineno",
        "message",
```